nilearn/datasets/tests/test_neurovault.py

```diff
@@ -803,9 +812,9 @@ def test_fetch_neurovault_ids(tmp_path):
     ].to_numpy()
 
     with pytest.raises(ValueError):
-        neurovault.fetch_neurovault_ids(mode="bad")
+        fetch_neurovault_ids(mode="bad")
```

Comment:
This test `test_fetch_neurovault_ids` is pretty long too, maybe it could be shortened? If not, I think a docstring and/or a few comments would help. 

I read it a few times but couldn't tell if the different sections are independent of each other or not (if they are dependent then they can't separated in different tests).

Reply:
yeah that one stuck out for me as well

and I had the same problem not knowing if some parts of it can be extracted

will give it a go

Reply:
looks like it could be split a bit